gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -538,6 +539,8 @@ public void initialize(URI path, Configuration config, boolean initSuperclass)
     logger.atFine().log(
         "initialize(path: %s, config: %s, initSuperclass: %b)", path, config, initSuperclass);
 
+    config =
+        ProviderUtils.excludeIncompatibleCredentialProviders(config, GoogleHadoopFileSystem.class);
```

Comment:
updating a method method parameter is considered a code smell. Can we treat the parameter as final and create a separate variable.

Also it might be a good idea to log the updated config.

Reply:
I see your point, but in this case, it's actually useful to overwrite `config` with the new modified instance. It prevents any chance of later code accidentally using the unmodified configuration, which could reintroduce bugs.

The difference in the modified `config` will be that that property `hadoop.security.credential.provider.path` has been removed, so there won't be much difference in the string representation.